### ISSUE TITLE
MD053 false positive for links in admonitions (MkDocs)

### ISSUE THREAD

**Issue by bittner** [markdownlint/markdownlint#474]
## Description
When you use a link in an admonition markdownlint doesn't recognize it's being used and complains falsely.

```markdown
!!! note

    Some linters [will complain][ruff] about a string value.

[ruff]: https://docs.astral.sh/ruff/rules/pytest-parametrize-names-wrong-type/
```

When you run markdownlint over a document with that code, it will complain, e.g.

```shell
...
pytest.en.md:167:1 MD053/link-image-reference-definitions Link and image reference definitions should be
needed [Unused link or image reference definition: "ruff"] [Context: "[ruff]: https://docs.astral.sh/..."]
```

### Environment

**MDL Version** v0.37.0 (included in MegaLinter v7)

### Expected Behavior
Markdownlint should recognize that the link is rightfully resolved, and not complain.

### Actual Behavior
Markdownlint complains (as shown above) even though the link is correctly used. This is also confirmed by the resulting HTML document, generated from the MarkDown code.

## Replication Case
1. Create a MkDocs project
1. Use an admonition in a document with a link (as shown above)
1. Run `markdownlint` over the code base

**Comment by DavidAnson** [comment]
From the way that error message looks, you are not using the Ruby implementation (this project), but the Node implementation over here: https://github.com/DavidAnson/markdownlint

If so, admonitions are not part of the CommonMark specification and not supported. Therefore, the syntax above of using four spaces to indent the text put it into a code block which means links are not converted and therefore the warning is correct.

This topic has been raised before: https://github.com/DavidAnson/markdownlint/issues/207

**Comment by jaymzh** [comment]
Closing, wrong project.
